src/jesse_state.erl

```diff
@@ -173,8 +173,19 @@ set_allowed_errors(#state{} = State, AllowedErrors) ->
 -spec set_current_schema( State :: state()
                         , NewSchema :: jesse:schema()
                         ) -> state().
-set_current_schema(#state{id = Id} = State, NewSchema) ->
-  NewId = combine_id(Id, jesse_json_path:value(?ID, NewSchema, undefined)),
+set_current_schema(#state{id = Id} = State, NewSchema0) ->
+  NewSchema =
+    case jesse_json_path:value(?REF, NewSchema0, undefined) of
+      undefined ->
+        NewSchema0;
+      Ref ->
+        %% Instead of just removing all the other fields, we put schema as
+        %% 1st element, so, only `ref' will be validated, while other fields
+        %% (say, `definitions') may still be referenced.
+        [{?REF, Ref} | lists:keydelete(?REF, 1, NewSchema0)]
```

Comment:
This looks like it crashes when the schema is a `map`. We caught this today before shipping some code to production... is there a reason not to support map format schemas here? It seems to work elsewhere.

Reply:
No, this was not intentional. I'll make a fix

Reply:
Hi @nalundgaard, may you please check that fix from https://github.com/for-GET/jesse/pull/113 works for you?